packages/ember-data/lib/system/model/model.js

```diff
@@ -1054,13 +656,11 @@ var Model = Ember.Object.extend(Ember.Evented, {
     successfully or rejected if the adapter returns with an error.
   */
   save: function() {
-    var promiseLabel = "DS: Model#save " + this;
-    var resolver = Ember.RSVP.defer(promiseLabel);
-
-    this.store.scheduleSave(this, resolver);
-
+    var model = this;
     return PromiseObject.create({
-      promise: resolver.promise
+      promise: this._internalModel.save().then(function() {
```

Comment:
if you call it inside ED you want to be dealing with internalModels, and not materialized versions, and only expose materialized ones at the boundaries. `save` is kinda special are you guaranteed to have the materializer record, but better to be consistent.